database-commons/src/main/java/io/cdap/plugin/db/config/AbstractDBSpecificSourceConfig.java

```diff
@@ -54,6 +55,17 @@ public abstract class AbstractDBSpecificSourceConfig extends PluginConfig implem
   @Description(Constants.Reference.REFERENCE_NAME_DESCRIPTION)
   public String referenceName;
 
+  @Name(PROPERTY_IMPORT_QUERY_TYPE)
```

Comment:
now that you have added import query type, shouldn't all checks be based on this. If importQuery is table, call table method otherwise query one